Show Jetpack badge on Me, App Settings, Activity Detail, Notifications Settings & Reader Post

ovitrif opened the pull request:
Shows the Jetpack Powered Badge in the WP app when the `JetpackPoweredFeatureConfig` flag is enabled on the following screens:
- Me
- App Settings
- Activity Detail
- Notifications Settings
- Reader Post

To test:
**WP App**
1. Launch **WP** app from PR build or local build of PR branch
2. Make sure the `JetpackPoweredFeatureConfig` flag is **on**
   > Me → App Settings → Debug Settings → check `JetpackPoweredFeatureConfig` & tap `RESTART THE APP`
3. Tap `Me` (profile icon in top right corner)
   ☑️ **Expect** the Jetpack powered badge to be **visible**
4. Tap `App Settings` and scroll to the bottom of the screen
   ☑️ **Expect** the Jetpack powered badge to be **visible**
5. Go back 2 times, now on the `MENU` tab of My Site tap `Activity Log` then tap any activity log item
   ⚠️ **Note:** _make sure to have at least one activity: if you don't, you can, for example, change your site title from Site Settings to generate one._
   ☑️ **Expect** the Jetpack powered badge to be **visible**
6. Go back 2 times, now tap `Notifications` (bottom right corner) then tap the `⚙️` (top right corner) to open `Notifications Settings` and scroll to the bottom of the screen
  ☑️ **Expect** the Jetpack powered badge to be **visible**
7. Go back, tap `Reader` (bottom middle) then tap any post to open `Reader Post view` and scroll to the bottom of the posts
  ☑️ **Expect** the Jetpack powered badge to be **visible** below the comments section
8. Make sure the `JetpackPoweredFeatureConfig` flag is **off**
9. Repeat steps 3-7 ☑️ **expecting** the Jetpack powered badge to **not** be visible

**JP App**
1. Launch **JP** app from PR build or local build of PR branch
2. Repeat steps 2-7 from `WP App` section ☑️ **expecting** the Jetpack powered badge to **not** be visible

## Screenshots
| Me (Night) | App Settings | Activity Detail | Notifications Settings | Reader Post |
| --- | --- | --- | --- | --- |
| <img width="130" src="https://user-images.githubusercontent.com/4588074/179174141-9d41cd75-1fe1-4806-b653-b393f85aa4e0.png"> | <img width="130" src="https://user-images.githubusercontent.com/4588074/179174198-c3e935b0-8164-45d3-b07b-d84c2faacf0e.png"> | <img width="130" src="https://user-images.githubusercontent.com/4588074/179174246-a60e9e5d-f0a4-40f4-b10d-4bc133eb8ac2.png"> | <img width="130" src="https://user-images.githubusercontent.com/4588074/179174305-6ff7cd24-e448-4386-975d-5fe3517e63a8.png"> | <img width="130" src="https://user-images.githubusercontent.com/4588074/179294985-68e9eaf5-96e0-4ab5-b0e9-a1cdda8cce8b.png"> |

### Review notes
- One reviewer should be enough to merge 👌 .

## Regression Notes
1. Potential unintended areas of impact
   Any of the screens affected.
3. What I did to test those areas of impact (or what existing automated tests I relied on)
   Tested manually.
9. What automated tests I added (or what prevented me from doing so)
   None, the changes are in code which is not testable with unit tests, and we're not adding UI tests for small UI changes.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

ovitrif committed:
Show jetpack badge on activity detail screen

ovitrif committed:
Show jetpack badge on me screen

ovitrif committed:
Show jetpack badge on app settings screen

ovitrif committed:
Add consistent bottom spacing to jetpack badge footer

ovitrif committed:
Rename jetpack_badge_bottom_spacing to jetpack_badge_footer_bottom_spacing

ovitrif committed:
Add jetpack badge to reader post detail view

ovitrif committed:
Update conditions for jetpack badge visibility to include feature flag

ovitrif committed:
Remove unused methods from reader post list fragment

ovitrif committed:
Use feature flag to show jetpack badge on app settings screen

ovitrif committed:
Show jetpack badge on notifications settings screen

irfano commented on WordPress/src/main/res/layout/jetpack_badge_footer.xml:
```diff
@@ -0,0 +1,13 @@
+<?xml version="1.0" encoding="utf-8"?>
+<FrameLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    android:id="@+id/jetpack_badge_footer"
+    android:layout_width="match_parent"
+    android:layout_height="wrap_content"
+    android:paddingBottom="@dimen/jetpack_badge_footer_bottom_spacing">
+
+    <include
+        android:id="@+id/jetpack_badge"
+        layout="@layout/jetpack_badge"
+        android:layout_width="match_parent"
+        android:layout_height="wrap_content" />
+</FrameLayout>
```
I couldn't understand why we need `jetpack_badge_footer` here. It looks like it's for being able to use `paddingBottom`. I know `paddingBottom` is not working in `include` But we can use `marginBottom` as you did in `reader_fragment_post_detail`, then we can remove jetpack_badge_footer. WDYT?

ovitrif replied:
The main reason for creating a separate layout file is because we're adding this component programmatically as footer view in the preferences list views of App Settings ([source](https://github.com/wordpress-mobile/WordPress-Android/blob/e65458e5810f57350c58d99f8b038b0985377c48/WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java#L249)) and Notification Settings([source](https://github.com/wordpress-mobile/WordPress-Android/blob/68c9c0b1478972dc6536143dc83ec00a28d1faf2/WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsFragment.java#L230)).

In both cases the padding is the same, hence my reasoning that it's cleaner with another XML than by programatically adding the margin 2 times.

If it would have been only to add padding and we could have used an `<include>` I agree that would be cleaner 👍 .

If your question was more about why we need the wrapping `FrameLayout`, I think It doesn't work without it.

Thank you for raising the question 🙇 .

irfano replied:
> we're adding this component programmatically as footer view in the preferences list views of App Settings ([source](https://github.com/wordpress-mobile/WordPress-Android/blob/e65458e5810f57350c58d99f8b038b0985377c48/WordPress/src/main/java/org/wordpress/android/ui/prefs/AppSettingsFragment.java#L249)) and Notification Settings([source](https://github.com/wordpress-mobile/WordPress-Android/blob/68c9c0b1478972dc6536143dc83ec00a28d1faf2/WordPress/src/main/java/org/wordpress/android/ui/prefs/notifications/NotificationsSettingsFragment.java#L230)).

I searched `jetpack_badge_footer` and saw it's being used only in `me_fragment`. I didn't notice they're also being used in App Settings and Notification Settings. Now that makes sense. 😄
Good work! 👍🏻

irfano reviewed (changes requested):
Nice work, Ovi!
- I added just a tiny comment about `jetpack_badge_footer`.
- I saw you also added the badge to the reader post detail screen. You can consider adding it to the PR description too.

ovitrif commented:
> * I saw you also added the badge to the reader post detail screen. You can consider adding it to the PR description too.

Thank you for noticing this @irfano! I had the screenshots saved but then when reverting the badge work for reader empty search I deleted all the reader screenshots and forgot about the implementation work done there while creating the PR 😅.

I updated the PR description and added a screenshot for that 👌 .

irfano reviewed (approved):
LGTM! 👍🏻
